internal/pkg/fieldtags/analyzer.go

```diff
@@ -61,11 +61,11 @@ var Analyzer = &analysis.Analyzer{
 		inspect.Analyzer,
 	},
 	ResultType: reflect.TypeOf(new(ResultType)).Elem(),
+	FactTypes:  []analysis.Fact{new(isTaggedField)},
```

Comment:
Are you using `Fact`s just as an organizational / testing tool?  This imposes a import-ordering and doesn't buy us much otherwise.  What issues in #68 were you hitting that requires this?

Reply:
We need to have some way to carry information about tagged fields over the course of the analysis. Consider:
```
package core

type Source struct {
  field string `levee:"source"`
}
```

```
package usescore

import "core"

func main() {
  f := core.Source{}.field
}
```

Under the current implementation, while the `source` analyzer is running on the `usescore` package, it does not receive any tagged fields from the `fieldtags` analyzer, because the tagged fields are in `core`, not `usescore`.

Typically, analyzer `Results` are the results for the analysis performed on a specific package. However, in this case we need to carry which tagged fields are sources to the importing packages. Otherwise, when a tagged field is accessed in an importing package, we have no way to tell.

It seems to me the most straightforward way to do this is to use `Fact`s. We are using this pattern in the `fieldpropagators` analyzer. The way it is being done right now is rather inefficient. It would be more efficient to use `PackageFact`s instead of `ObjectFact`s. That way we could bundle up all the tagged fields into a slice or map, and moving it forward from importee to importer would be very efficient. Right now I don't see this hindering performance significantly, and without a benchmark I think it is preferable to use `ObjectFact`s since the code is simpler that way.